homeassistant/components/asuswrt/router.py

```diff
@@ -349,9 +362,14 @@ async def init_sensors_coordinator(self) -> None:
             SENSORS_TYPE_COUNT: SENSORS_CONNECTED_DEVICE,
             SENSORS_TYPE_LOAD_AVG: SENSORS_LOAD_AVG,
             SENSORS_TYPE_RATES: SENSORS_RATES,
+            SENSORS_TYPE_TEMPERATURES: SENSORS_TEMPERATURES,
         }
 
         for sensor_type, sensor_names in sensors_types.items():
+            if sensor_type == SENSORS_TYPE_TEMPERATURES:
+                sensor_names = await self._get_available_temperature_sensors()
```

Comment:
Why are we fetching the sensor's inside this loop? We don't seem to vary the input to this method depending on the loop item.

Reply:
Number of temperature sensors on asus routers vary depending on number of radio, and in some routers are not available at all

Reply:
Right, but we are calling the same function multiple times. I don't see what's different each time we call this function.

Reply:
Why multiple times? It should be called only ones at startup to check type `SENSORS_TYPE_TEMPERATURES`.

Reply:
I see now that we only do this for the `SENSORS_TYPE_TEMPERATURES` key in the loop of the dict. We could move this call out of the loop and populate the dict item with the result from the beginning. That may be a bit more clear and we would need the check for which key to to this call for. We can still keep the check for `sensor_names` below.

Reply:
Ok, agree with you.
@Chen-IL, could you create a follow up PR to apply @MartinHjelmare suggestion?

Reply:
#60975